arch/arm/core/aarch32/isr_wrapper.S

```diff
@@ -162,6 +165,15 @@ _idle_state_cleared:
 #endif /* CONFIG_EXECUTION_BENCHMARKING */
 	blx r3		/* call ISR */
 
+#if defined(CONFIG_CPU_CORTEX_A) || defined(CONFIG_CPU_CORTEX_R)
+	pop {r0, r1}
+#if !defined(CONFIG_ARM_CUSTOM_INTERRUPT_CONTROLLER)
+	bl arm_gic_eoi
+#else
+	bl z_soc_irq_eoi
+#endif /* !CONFIG_ARM_CUSTOM_INTERRUPT_CONTROLLER */
+#endif /* CONFIG_CPU_CORTEX_A || CONFIG_CPU_CORTEX_R */
+
```

Comment:
Adding the comment here, because I can't write below:
Is the block starting in line 186 also applicable for CORTEX_A? Since you are refactoring this part, it is ok to do this extra work now, and be proactive for Cortex-A :)

Reply:
Same for line 197. 
Note: if this refactoring cannot make the code work _perfectly_ for Cortex-A, I suggest we stick to Cortex-R only..

Reply:
> Is the block starting in line 186 also applicable for CORTEX_A? Since you are refactoring this part, it is ok to do this extra work now, and be proactive for Cortex-A :)

Yes, almost all the Cortex-R-specific code will be applicable to the Cortex-A as well.

https://github.com/ibirnbaum/zephyr/blob/7fd72d0d6686e57e30af99e612f97a416da69697/arch/arm/core/aarch32/isr_wrapper.S#L178

Reply:
> Note: if this refactoring cannot make the code work _perfectly_ for 32-bit Cortex-A, I suggest we stick to Cortex-R only..

I guess I could try, but it seems that would be outside the scope of this PR.

For now, I will drop all the Cortex-A checks in this PR for the sake of consistency (we will need to remember to add them back in when we add the Cortex-A port).

Reply:
Great

Reply:
Dropped all Cortex-A checks for now